### DISCUSSION TITLE
Help Setting Up Neovim

### DISCUSSION THREAD

**Discussion by illogicalbingo** [phpactor/phpactor#2696] [Q&A]
I am very new to PHP, so please forgive my ignorance.

I am having an issue were phpactor reports undefined variables for variables that are required from other php docs
![image](https://github.com/user-attachments/assets/aaa8a828-99be-425a-a567-d0a144efdc0c)

I have no idea what I am doing wrong, I have tried installing phpactor through mason and directly on my host machine, both options yield the same results.

This is my lsp config
```lua
		local lspconfig = require("lspconfig")
		lspconfig.phpactor.setup({
			settings = {
				["phpactor"] = {
					on_attach = on_attach,
					capabilities = capabilities,
				},
			},
		})
```

Any help / directions to were I can find help would be appreciated.

**Comment by dantleech** [comment]
This is "expected" behavior currently. Phapctor does not analyse files included via. `require` and it's been developed with more of an expectation that the project is using [composer](https://getcomposer.org/download/) and doesn't rely on `require` and `include`.

At one point Phpactor _did_ process these files, but it caused some serious performance issues.

**Comment by illogicalbingo** [comment]
Hi Dan,
Thanks for the swift reply. 

I guess I will need to educate myself on compose and how to use that instead of require and include, might I ask is using composer the preferred practice in general?
